ncm-network/src/main/perl/network.pm

```diff
@@ -325,7 +325,14 @@ sub file_dump
     # Make hardlinked copy
     #   hardlink is ok, new file will be put in place, no editing of existing file
     if (!$self->file_exists($file) || $self->mk_bu($file, $testcfg)) {
-        my $fh = CAF::FileWriter->new($testcfg, log => $self, keeps_state => 1);
+        my $fh;
+        # files in nmconnection has to be mode 0400 otherwise networkmanager won't read it.
+        if ($file =~ /\.nmconnection$/) {
```

Comment:
hmm, i don't like copy paste. 
can you test

```
# files in nmconnection has to be mode 0400 otherwise networkmanager won't read it.
my $mode;
$mode = 0400 if ($file =~ /\.nmconnection$/);


my $fh = CAF::FileWriter->new($testcfg, log => $self, keeps_state => 1, mode => $mode);
```